tests/Unit/Core/Util/ConfigurationVariablesLoaderTest.php

```diff
@@ -0,0 +1,182 @@
+                    'mailer_user' => 'myuser@gmail.com',
+                    'mailer_password' => 'thisismymailerpassword',
+                    'locale' => 'ar-AR',
+                    'secret' => 'this1s4paSspHraS3',
+                    'ps_caching' => 'CacheMemcache',
+                    'ps_cache_enable' => 'true',
+                    'use_debug_toolbar' => 'true',
+                ],
+            ], $configurationVariablesLoader->loadEnvironmentVariables(['parameters' => []]));
+    }
+
+    public function testLoadEnvironmentVariablesReplacedAllowedParameters(): void
+    {
+        $envFile = $this->createEnvFile('PS_DATABASE_PORT=33123
+PS_DATABASE_NAME=test_database
+PS_DATABASE_PASSWORD=random_pwd
+PS_DATABASE_PREFIX=test
+PS_DATABASE_ENGINE=postgresql
+PS_SOMETHING_NOT_ALLOWED=fake_value
+PS_MAILER_TRANSPORT=sendmail
+PS_MAILER_HOST=127.0.0.1
+PS_MAILER_USER=toto
+PS_LOCALE=fr-FR
+PS_CACHE_ENABLE=false
+');
+
+        $configurationVariablesLoader = new ConfigurationVariablesLoader($envFile);
+
+        $this->assertSame(
+            [
+                'parameters' => [
+                    'database_host' => '192.168.1.1',
+                    'database_port' => '33123',
+                    'database_user' => 'superadmin',
+                    'locale' => 'fr-FR',
+                    'database_password' => 'random_pwd',
+                    'database_name' => 'test_database',
+                    'database_prefix' => 'test',
+                    'database_engine' => 'postgresql',
+                    'mailer_transport' => 'sendmail',
+                    'mailer_host' => '127.0.0.1',
+                    'mailer_user' => 'toto',
+                    'ps_cache_enable' => 'false',
+                ],
+            ],
+            $configurationVariablesLoader->loadEnvironmentVariables(
+                [
+                    'parameters' => [
+                        'database_host' => '192.168.1.1',
+                        'database_port' => '32768',
+                        'database_user' => 'superadmin',
+                        'locale' => 'en-US',
+                    ],
+                ]
+            )
+        );
+    }
+
+    private function createBasicEnvFile()
+    {
+        return $this->createEnvFile('PS_DATABASE_HOST=localhost
+PS_DATABASE_PORT=3306
+PS_DATABASE_NAME=test_db
+PS_DATABASE_USER=db_user
+PS_DATABASE_PASSWORD=db_password
+PS_DATABASE_PREFIX=ps
+PS_DATABASE_ENGINE=mysql
+PS_COOKIE_KEY=myCookie
+PS_COOKIE_IV=myCookieIV
+PS_NEW_COOKIE_KEY=myNewCookieKey
+PS_MAILER_TRANSPORT=smtp
+PS_MAILER_HOST=smtp.gmail.com
+PS_MAILER_USER=myuser@gmail.com
+PS_MAILER_PASSWORD=thisismymailerpassword
+PS_LOCALE=ar-AR
+PS_SECRET=this1s4paSspHraS3
+PS_CACHING=CacheMemcache
+PS_CACHE_ENABLE=true
+PS_USE_DEBUG_TOOLBAR=true
+');
+    }
+
+    private function createEnvFile(string $content): string
+    {
+        $root = vfsStream::setup();
+
+        $envFile = vfsStream::newFile('.env', 0400)->at($root);
```

Comment:
I guess this file needs to be removed at the end of the test?

Reply:
Yep, this test should be reviewed to manage filesystem properly I guess :thinking: